views/analysis.py

```diff
@@ -572,6 +572,111 @@ def validate_analysis_extent(request):
         return HttpResponseServerError()
 
 
+def calculate_area(request, hazard_id, exposure_id, view_extent):
```

Comment:
This method is almost a duplicate of validate_analysis_extent.
You should try to improve that one instead of creating another one.
In favor of DRY Principle: https://en.wikipedia.org/wiki/Don%27t_repeat_yourself